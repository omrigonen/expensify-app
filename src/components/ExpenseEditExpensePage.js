import React from 'react';

const ExpenseEditExpensePage = (props) => {
        console.log(props);
        return (
        <div>
            Editing the expense id of {props.match.params.id}
        </div>
    );
}

export default ExpenseEditExpensePage;