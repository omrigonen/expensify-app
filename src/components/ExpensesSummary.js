
import React from 'react';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import { connect } from 'react-redux';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
    const total = numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00');
    const count = props.expenses.length;
    return (
        <div>
            <h1>Viewing {count} expense{count === 1 ? '' : 's'} totalling {total}</h1>
        </div>
    )
};

const mapStateToProps = (state) => (
    {
        expenses: selectExpenses(state.expenses, state.filters)
    });

export default connect(mapStateToProps)(ExpensesSummary);