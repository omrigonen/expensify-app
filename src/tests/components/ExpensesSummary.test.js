import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';

test('should render when totaling 3 items',() => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render when totaling 1 items',() => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render when totaling 0 items',() => {
    const wrapper = shallow(<ExpensesSummary expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});