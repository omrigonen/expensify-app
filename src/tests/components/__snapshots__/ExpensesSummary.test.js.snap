// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render when totaling 0 items 1`] = `
<div>
  <h1>
    Viewing 
    0
     expense
    s
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`should render when totaling 1 items 1`] = `
<div>
  <h1>
    Viewing 
    1
     expense
     totalling 
    $1.95
  </h1>
</div>
`;

exports[`should render when totaling 3 items 1`] = `
<div>
  <h1>
    Viewing 
    3
     expense
    s
     totalling 
    $57.90
  </h1>
</div>
`;
